controller/eks-cluster-config-handler.go

```diff
@@ -1326,6 +1326,12 @@ func createNodeGroup(eksConfig *eksv1.EKSClusterConfig, group eksv1.NodeGroup, e
 		},
 	}
 
+	if gpu := group.Gpu; aws.BoolValue(gpu) {
+		nodeGroupCreateInput.AmiType = aws.String(eks.AMITypesAl2X8664Gpu)
+	} else {
+		nodeGroupCreateInput.AmiType = aws.String(eks.AMITypesAl2X8664)
+	}
```

Comment:
opinionated nit: do not know how much we care about DRY, but you might be able to reduce this to...

```suggestion
      amiType := eks.AMITypesAl2X8664
	if aws.BoolValue(group.Gpu) {
		amiType = eks.AMITypesAl2X8664Gpu
	}
	nodeGroupCreateInput.AmiType  = aws.String(amiType)
```

Granted, this approach can have a scenario where there's two assignment calls, so it's a tradeoff.